packages/scandipwa/src/route/SendConfirmationPage/SendConfirmationPage.component.js

```diff
@@ -0,0 +1,146 @@
+        email: PropTypes.string,
+        redirect: PropTypes.bool.isRequired,
+        isLoading: PropTypes.bool.isRequired,
+        shouldDisplayWarning: PropTypes.bool.isRequired,
+        onConfirmSuccess: PropTypes.func.isRequired,
+        onFormError: PropTypes.func.isRequired
+    };
+
+    static defaultProps = {
+        email: ''
+    };
+
+    renderWarningMessage() {
+        const { shouldDisplayWarning } = this.props;
+
+        if (!shouldDisplayWarning) {
+            return null;
+        }
+
+        return (
+            <div block="SendConfirmationPage" elem="WarningMsg">
+                <h2>
+                    { __('Unable to re-request confirmation email') }
+                </h2>
+                <div>
+                    { __('The URL is invalid. Some parameters are missing.') }
+                </div>
+            </div>
+        );
+    }
+
+    renderForm() {
+        const {
+            email,
+            onConfirmSuccess,
+            onFormError
+        } = this.props;
+
+        return (
+            <Form
+              mix={ { block: 'SendConfirmationPage', elem: 'Form' } }
+              key="resend-confirmation"
+              onSubmit={ onConfirmSuccess }
+              onError={ onFormError }
+            >
+                <Field
+                  type={ FIELD_TYPE.email }
+                  label={ __('Email') }
+                  attr={ {
+                      id: 'email',
+                      name: 'email',
+                      defaultValue: email,
+                      placeholder: __('Your email name')
+                  } }
+                  validateOn={ ['onChange'] }
+                  validationRule={ {
+                      isRequired: true,
+                      inputType: VALIDATION_INPUT_TYPE.email
+                  } }
+                  addRequiredTag
+                />
+                <button
+                  type="submit"
+                  block="Button"
+                  mix={ { block: 'SendConfirmationPage', elem: 'Button' } }
+                >
+                    { __('Send confirmation link') }
+                </button>
+            </Form>
+        );
+    }
+
+    renderPageContents() {
+        const { shouldDisplayWarning } = this.props;
+
+        if (shouldDisplayWarning) {
+            return null;
+        }
+
+        return (
+            <>
+                <h1 block="SendConfirmationPage" elem="Heading">
+                    { __('Send confirmation link') }
+                </h1>
+                <p block="SendConfirmationPage" elem="SubHeading">
+                    { __('Please enter your email below and we will send you the confirmation link.') }
+                </p>
+                { this.renderForm() }
+            </>
+        );
+    }
+
+    render() {
+        const {
+            redirect,
+            isLoading
+        } = this.props;
+
+        if (redirect || isSignedIn()) {
+            return <Redirect to="/" />;
```

Comment:
Will `<Redirect />` a history push action or a real redirect?

Reply:
This transition does not open the page again, like a normal redirect, but it does not save the path in history (that is, when click "back", the transition will go to the page that was opened before "/customer/account/confirmation")

Is it better to use history?

Reply:
No, it's ok since you return the component from the render method.
Just wanted to make sure it will rerender the page instead of reloading it.